docs/tutorial/writing-a-more-complex-dataset-definition/index.md

```diff
@@ -254,14 +254,31 @@ dataset.sex = patients.sex
 ```
 
 We call `patients.age_on` to determine the age of a patient on the index date
-and assign the result column to `dataset.age`.
+and assign the result column to the `age` variable.
 Remember that to be included in the population,
 a patient was an adult on the index date.
 
 ```python
-dataset.age = patients.age_on(index_date)
+age = patients.age_on(index_date)
 ```
 
+We then use the `age` variable to match the patient to one of five age
+bands using `case()` expression and assign this to `dataset.age_band`.
+
+```python
+dataset.age_band = case(
+    when((age >= 18) & (age < 50)).then("age_18_49"),
+    when((age >= 50) & (age < 65)).then("age_50_64"),
+    when((age >= 65) & (age < 75)).then("age_65_74"),
+    when((age >= 75) & (age < 85)).then("age_75_84"),
+    when((age >= 85)).then("age_85_plus"),
+)
+```
+
+The `case()` expression consists of a list of `when(<condition>).then(<value>)`
+expressions, where `<condition>` is always a Boolean and `<value>` can
+be any type as long as it's consistent with the `case()` expression.
```

Comment:
Is it obvious what we mean by "as long as it's consistent with the case() expression"? What would be inconsistent?

Reply:
I think I meant to write "within the case expression". Does that make enough sense on its own?